Src/Support/Google.Apis.Core/Util/Utilities.cs

```diff
@@ -68,6 +68,39 @@ internal static bool IsNullOrEmpty<T>(this IEnumerable<T> coll)
             return coll == null || coll.Count() == 0;
         }
 
+        /// <summary>
+        /// Checks that the given value is in fact defined in the enum used as the type argument of the method.
+        /// </summary>
+        /// <typeparam name="T">The enum type to check the value within.</typeparam>
+        /// <param name="value">The value to check.</param>
+        /// <param name="paramName">The name of the parameter whose value is being tested.</param>
+        /// <returns><paramref name="value"/> if it was a defined value</returns>
+        public static T ThrowIfUndefined<T>(this T value, string paramName) where T : struct
```

Comment:
I think I'd prefer this to be a regular method rather than an extension method, particularly as it's applicable (as far as the compiler is concerned) for *all* value types, but it's only useful for structs.
I'd also suggest putting the word "enum" in the name: `CheckEnumValue` as per GaxPreconditions would be fine, I think.